src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php

```diff
@@ -401,6 +404,90 @@ public function editAction(int $addressId, Request $request): Response
         ]);
     }
 
+    /**
+     * Handles edit form rendering and submission for order address
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_orders_index"
+     * )
+     *
+     * @param int $orderId
+     * @param string $addressType
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function editOrderAction(int $orderId, string $addressType, Request $request): Response
+    {
+        try {
+            $order = new Order($orderId);
+            $addressId = null;
+            switch ($addressType) {
+                case 'delivery':
+                    $addressType = OrderAddressType::DELIVERY_ADDRESS_TYPE;
+                    $addressId = $order->id_address_delivery;
+                    break;
+                case 'invoice':
+                    $addressType = OrderAddressType::INVOICE_ADDRESS_TYPE;
+                    $addressId = $order->id_address_invoice;
+                    break;
```

Comment:
default ? An exception ?

Reply:
The exception is already managed in the command class which checks allowed address types